Source/WTF/wtf/StackTrace.cpp

```diff
@@ -44,31 +45,27 @@ void WTFGetBacktrace(void** stack, int* size)
 
 namespace WTF {
 
-ALWAYS_INLINE size_t StackTrace::instanceSize(int capacity)
+ALWAYS_INLINE size_t StackTrace::instanceSize(int numberOfFrames)
 {
-    ASSERT(capacity >= 1);
-    return sizeof(StackTrace) + (capacity - 1) * sizeof(void*);
+    int numberOfExtraFrames = numberOfFrames - std::extent<decltype(StackTrace::m_stack)>::value;
+    return sizeof(StackTrace) + numberOfExtraFrames * sizeof(void*);
 }
 
 std::unique_ptr<StackTrace> StackTrace::captureStackTrace(int maxFrames, int framesToSkip)
 {
     maxFrames = std::max(1, maxFrames);
-    size_t sizeToAllocate = instanceSize(maxFrames);
-    std::unique_ptr<StackTrace> trace(new (NotNull, fastMalloc(sizeToAllocate)) StackTrace());
-
     // Skip 2 additional frames i.e. StackTrace::captureStackTrace and WTFGetBacktrace.
     framesToSkip += 2;
     int numberOfFrames = maxFrames + framesToSkip;
-
-    WTFGetBacktrace(&trace->m_skippedFrame0, &numberOfFrames);
-    if (numberOfFrames) {
-        RELEASE_ASSERT(numberOfFrames >= framesToSkip);
+    std::unique_ptr<StackTrace> trace(new (NotNull, fastMalloc(instanceSize(numberOfFrames))) StackTrace());
+    WTFGetBacktrace(trace->m_stack, &numberOfFrames);
+    if (numberOfFrames > framesToSkip) {
```

Comment:
Let's move this `WTFGetBacktrace` call inside the `if (numberOfFrames > framesToSkip)` check.  We don't need to call it if we don't intend to use any of the captured frames anyway.